src/internal/event-bridge.ts

```diff
@@ -0,0 +1,139 @@
+import { JSONObject } from 'k6'
+import http, { RefinedResponse, ResponseType } from 'k6/http'
+
+import { AWSClient } from './client'
+import { AWSConfig } from './config'
+import { AWSError } from './error'
+import { InvalidSignatureError, SignatureV4 } from './signature'
+import { AMZ_TARGET_HEADER } from './constants'
+import { HTTPHeaders, HTTPMethod } from './http'
+
+type PutEventEntry = {
+    Detail: string
+    DetailType: string
+    EventBusName: string
+    Resources: [string]
+    Source: string
+}
+
+type PutEventsInput = {
```

Comment:
I'm not opposed to using type definitions per-se. However, it's been our practice so far to use interface types instead when defining a "contract" object that's meant to represent function arguments.

I would suggest the following (including documentation):

```suggestion
/**
 * Represents the input for a put events operation.
 *
 * @typedef {Object} PutEventsInput
 *
 * @property {string} [EndpointId] - The optional URL subdomain of the endpoint.
 * @property {Partial<PutEventEntry>[]} Entries - An array of entries that defines an event in your system.
 */
interface PutEventsInput {
    EndpointId?: string
    Entries: Partial<PutEventEntry>[]
}
```

What do you think?